### PR TITLE
Rule S3264: add repro for #3931

### PR DESCRIPTION
**Pull request by csaba-sagi-sonarsource**
Reproduces #3931

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UninvokedEventDeclaration.cs [hunk 1]
```diff
@@ -204,5 +204,23 @@ public void InvokeAll()
                 outer.MyAction6?.Invoke(this, EventArgs.Empty);
             }
         }
+
+        // https://github.com/SonarSource/sonar-dotnet/issues/3931
+        public delegate void SomeDelegate(int a);
+
+        public class ReproForIssue3931
+        {
+            public event SomeDelegate ActuallyUsedEvent; // Noncompliant FP
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UninvokedEventDeclaration.cs:213]: I'm not sure I get it. Where is `ActuallyUsedEvent` used?

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]
Just a small question related to the usage of `ActuallyUsedEvent`

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UninvokedEventDeclaration.cs (hunk 1 above)
Comment (csaba-sagi-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UninvokedEventDeclaration.cs:213]: I messed it up, I`ll fix it. I tried to combine it with the example from the community and messed it up. Fix is on its way. Sorry...

Comment (costin-zaharia-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UninvokedEventDeclaration.cs:213]: No worries, it looks good now. 

I would add though an additional case also with a call of `.GetInvocationList();` on another event for which we don't do an iteration. We will have to decide if a call to `.GetInvocationList()` will "sanitize" the event or if we want to actually check for iterations and invoke calls.

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/src/SonarAnalyzer.CSharp/Rules/UninvokedEventDeclaration.cs [hunk 2]
```diff
@@ -38,62 +38,55 @@ public sealed class UninvokedEventDeclaration : SonarDiagnosticAnalyzer
...
-                .GetRemovableFieldLikeDeclarations(eventSyntax, maxAccessibility)
+                .GetRemovableFieldLikeDeclarations(EventSyntax, MaxAccessibility)
                 .ToList();
 
             if (!removableEventFields.Any())
             {
                 return;
             }
 
-            var invokedSymbols = GetInvokedEventSymbols(removableDeclarationCollector);
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/UninvokedEventDeclaration.cs:77]: we'd need to come up with a reproducer, but I suspect this change lead to the #4498 perf regression

**Pull request merged**